### PR TITLE
feat(onboarding): add suggested questions answer

### PR DESCRIPTION
**Pull request by mamadoudicko**
https://github.com/StanGirard/quivr/assets/63923024/08554dd5-2765-436a-b900-48b38d582f6c

https://github.com/StanGirard/quivr/issues/1389

### CODE REVIEW

File: frontend/app/chat/[chatId]/components/ActionsBar/components/ChatInput/components/OnboardingQuestions/components/OnboardingQuestion/hooks/useOnboardingQuestion.ts [hunk 1]
```diff
@@ -1,6 +1,9 @@
+import { useEffect, useState } from "react";
 import { useTranslation } from "react-i18next";
 
-import { useChat } from "@/app/chat/[chatId]/hooks/useChat";
+import { useStreamText } from "@/app/chat/[chatId]/components/ChatDialogueArea/components/ChatDialogue/components/Onboarding/hooks/useStreamText";
```
Comment (gozineb) [inline frontend/app/chat/[chatId]/components/ActionsBar/components/ChatInput/components/OnboardingQuestions/components/OnboardingQuestion/hooks/useOnboardingQuestion.ts:4]: Maybe it's time to move `useStreamText` elsewhere more accessible for all the components using it

File: frontend/app/chat/[chatId]/components/ActionsBar/components/ChatInput/components/OnboardingQuestions/components/OnboardingQuestion/hooks/useOnboardingQuestion.ts [hunk 2]
```diff
@@ -10,18 +13,34 @@ import { questionIdToTradPath } from "../utils";
...
+
+  const { lastStream } = useStreamText({
+    text: t(`onboarding.answer.${onboardingStep}`),
+    enabled: isAnswerRequested,
+  });
+
+  useEffect(() => {
+    if (isAnswerRequested) {
+      const chatMessage: ChatMessage = {
+        chat_id: questionId,
```
[the remark this answers is not in the document]
Comment (gozineb) [reply frontend/app/chat/[chatId]/components/ActionsBar/components/ChatInput/components/OnboardingQuestions/components/OnboardingQuestion/hooks/useOnboardingQuestion.ts:30]: You would also need to create a new chat first in order to get a chatId - before updating the chat with the question + answer

File: frontend/app/chat/[chatId]/components/ActionsBar/components/ChatInput/components/OnboardingQuestions/components/OnboardingQuestion/hooks/useOnboardingQuestion.ts [hunk 3]
```diff
@@ -10,18 +13,34 @@ import { questionIdToTradPath } from "../utils";
...
+      void updateStreamingHistory(chatMessage);
+    }
+  }, [isAnswerRequested, question, questionId, lastStream]);
 
   const handleSuggestionClick = async () => {
-    await Promise.all([
-      addQuestion(question),
-      updateOnboarding({ [questionId]: false }),
-    ]);
+    setIsAnswerRequested(true);
```
Comment (gozineb) [inline frontend/app/chat/[chatId]/components/ActionsBar/components/ChatInput/components/OnboardingQuestions/components/OnboardingQuestion/hooks/useOnboardingQuestion.ts:42]: Here replace "addQuestion(question)" with either create new chat + update chat or with update chat only if the question is asked in an already existing chat

File: frontend/public/locales/en/chat.json
Comment (gozineb) [inline frontend/public/locales/en/chat.json:1]: I think we'll soon have to tranform this .json in .ts in order to user the CMS within, wdyt ?

Comment (mamadoudicko) [reply frontend/public/locales/en/chat.json:1]: Why ?

File: frontend/app/chat/[chatId]/components/ActionsBar/components/ChatInput/components/OnboardingQuestions/components/OnboardingQuestion/hooks/useOnboardingQuestion.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gozineb) [reply frontend/app/chat/[chatId]/components/ActionsBar/components/ChatInput/components/OnboardingQuestions/components/OnboardingQuestion/hooks/useOnboardingQuestion.ts:30]: Nop have /chat update route where you can update the whole chat with old chat + question + answer from onboarding_b

**Review by gozineb — APPROVED** [review]

**Pull request merged**